### PR TITLE
[DataFactory]Added new features into 8.0.0

### PR DESCRIPTION
**Pull request by Jingshu923**
- Added properties to spark job activity https://github.com/Azure/azure-rest-api-specs/pull/20879
- Added disablePublish property into FactoryRepoConfiguration  https://github.com/Azure/azure-rest-api-specs/pull/21095
- Added scriptBlockExecutionTimeout property into Script Activity https://github.com/Azure/azure-rest-api-specs/pull/21255

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/datafactory/Microsoft.Azure.Management.DataFactory/src/Customizations/FactoryVSTSConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        /// Initializes a new instance of the FactoryVSTSConfiguration class.
+        /// </summary>
+        /// <param name="accountName">Account name.</param>
+        /// <param name="repositoryName">Repository name.</param>
+        /// <param name="collaborationBranch">Collaboration branch.</param>
+        /// <param name="rootFolder">Root folder.</param>
+        /// <param name="projectName">VSTS project name.</param>
+        /// <param name="lastCommitId">Last commit id.</param>
+        /// <param name="tenantId">VSTS tenant id.</param>
+        public FactoryVSTSConfiguration(string accountName, string repositoryName, string collaborationBranch, string rootFolder, string projectName, string lastCommitId = default(string), string tenantId = default(string))
```
Comment (Yao725) [inline sdk/datafactory/Microsoft.Azure.Management.DataFactory/src/Customizations/FactoryVSTSConfiguration.cs:15]: I don't think it's a right way to minimize the breaking change. If the customer calls this method in the previous version like:
```csharp
var test = FactoryVSTSConfiguration("accountName", "repositoryName", "collaborationBranch", "rootFolder", "projectName")
```
There is no doubt that this call will become an ambiguous call in the new version. Therefore I think a more appropriate customization should be:
```csharp
public FactoryVSTSConfiguration(string accountName, string repositoryName, string collaborationBranch, string rootFolder, string projectName, string lastCommitId, string tenantId)
```

File: sdk/datafactory/Microsoft.Azure.Management.DataFactory/src/Customizations/ScriptActivity.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        /// <param name="description">Activity description.</param>
+        /// <param name="dependsOn">Activity depends on condition.</param>
+        /// <param name="userProperties">Activity user properties.</param>
+        /// <param name="linkedServiceName">Linked service reference.</param>
+        /// <param name="policy">Activity policy.</param>
+        /// <param name="scripts">Array of script blocks. Type: array.</param>
+        /// <param name="logSettings">Log settings of script activity.</param>
+        public ScriptActivity(string name, IDictionary<string, object> additionalProperties = default(IDictionary<string, object>), string description = default(string), IList<ActivityDependency> dependsOn = default(IList<ActivityDependency>), IList<UserProperty> userProperties = default(IList<UserProperty>), LinkedServiceReference linkedServiceName = default(LinkedServiceReference), ActivityPolicy policy = default(ActivityPolicy), IList<ScriptActivityScriptBlock> scripts = default(IList<ScriptActivityScriptBlock>), ScriptActivityTypePropertiesLogSettings logSettings = default(ScriptActivityTypePropertiesLogSettings))
```
Comment (Yao725) [inline sdk/datafactory/Microsoft.Azure.Management.DataFactory/src/Customizations/ScriptActivity.cs:20]: Same as above, we should change the parameters to be mandatory that are before the newly inserted parameter and also the parameter which has the same location as the new parameter. In thie ctor, the parameters from `additionalProperties` to `scripts` should not be optional parameter anymore.

File: sdk/datafactory/Microsoft.Azure.Management.DataFactory/src/Customizations/SynapseSparkJobDefinitionActivity.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        /// resultType string).</param>
+        /// <param name="numExecutors">Number of executors to launch for this
+        /// job, which will override the 'numExecutors' of the spark job
+        /// definition you provide.</param>
+        public SynapseSparkJobDefinitionActivity(string name, SynapseSparkJobReference sparkJob, IDictionary<string, object> additionalProperties = default(IDictionary<string, object>), string description = default(string), IList<ActivityDependency> dependsOn = default(IList<ActivityDependency>), IList<UserProperty> userProperties = default(IList<UserProperty>), LinkedServiceReference linkedServiceName = default(LinkedServiceReference), ActivityPolicy policy = default(ActivityPolicy), IList<object> arguments = default(IList<object>), object file = default(object), object className = default(object), IList<object> files = default(IList<object>), BigDataPoolParametrizationReference targetBigDataPool = default(BigDataPoolParametrizationReference), object executorSize = default(object), object conf = default(object), object driverSize = default(object), int? numExecutors = default(int?))
```
[the remark this answers is not in the document]
Comment (Jingshu923) [reply sdk/datafactory/Microsoft.Azure.Management.DataFactory/src/Customizations/SynapseSparkJobDefinitionActivity.cs]: fix

File: sdk/datafactory/Microsoft.Azure.Management.DataFactory/src/Customizations/ScriptActivity.cs (hunk 2 above)
Comment (Jingshu923) [reply sdk/datafactory/Microsoft.Azure.Management.DataFactory/src/Customizations/ScriptActivity.cs]: fix

File: sdk/datafactory/Microsoft.Azure.Management.DataFactory/src/Customizations/FactoryVSTSConfiguration.cs (hunk 1 above)
Comment (Jingshu923) [reply sdk/datafactory/Microsoft.Azure.Management.DataFactory/src/Customizations/FactoryVSTSConfiguration.cs]: fix

**Review by Yao725 — APPROVED** [review]
LGTM

**Pull request merged**